sympy/physics/mechanics/joint.py

```diff
@@ -0,0 +1,356 @@
+        if not ax.dt(body.frame) == 0:
+            raise ValueError('Axis cannot be time-varying.')
+        return ax
+
+    def _locate_joint_pos(self, body, joint_pos):
+        if joint_pos is None:
+            joint_pos = Vector(0)
+        if not isinstance(joint_pos, Vector):
+            raise ValueError('Joint Position must be supplied as Vector.')
+        if not joint_pos.dt(body.frame) == 0:
+            raise ValueError('Position Vector cannot be time-varying.')
+        return body.masscenter.locatenew(self._name + '_' + body.name + '_joint', joint_pos)
+
+    def _align_axis(self, parent, child):
+        # Returns the axis and angle between two axis(vectors).
+        angle = parent.angle_between(child)
+        axis = cross(child, parent).normalize()
+        return angle, axis
+
+
+class PinJoint(Joint):
+    """Pin (Revolute) Joint.
+
+    Explanation
+    ===========
+
+    It is defined such that the child body rotates with respect to the parent body
+    about the body fixed parent axis through the angle theta. The point of joint
+    (pin joint) is defined by two points in each body which coincides together.
+    In addition, the child's reference frame can be arbitrarily rotated a constant
+    amount with respect to the parent axis by passing an axis in child body which
+    must align with parent axis after the rotation.
+
+    Parameters
+    ==========
+
+    name : string
+        The Joint's name which makes it unique.
+    parent : Body
+        The parent body of joint.
+    child : Body
+        The child body of joint.
+    coordinates: dynamicsymbol, optional
+        Coordinates of joint.
+    speeds : dynamicsymbol, optional
+        Speeds of joint.
+    parent_joint_pos : Vector, optional
+        Defines the joint's point where the parent will be connected to child.
+        Default value is masscenter of Parent Body.
+    child_joint_pos : Vector, optional
+        Defines the joint's point where the child will be connected to parent.
+        Default value is masscenter of Child Body.
+    parent_axis : Vector, optional
+        Axis of parent frame which would be be aligned with child's
+        axis. Default is x axis in parent's frame.
+    child_axis : Vector, optional
+        Axis of child frame which would be aligned with parent's
+        axis. Default is x axis in child's frame.
+
+    Examples
+    =========
+
+    This is minimal working example where parent body and child body is
+    connected via PinJoint through their masscenters.
+
+        >>> from sympy.physics.mechanics import Body, PinJoint
+        >>> parent = Body('parent')
+        >>> child = Body('child')
+        >>> P = PinJoint('P', parent, child)
+        >>> P.coordinates()
+        [theta_P(t)]
+        >>> P.speeds()
+        [omega_P(t)]
+        >>> P.kdes()
+        [omega_P(t) - Derivative(theta_P(t), t)]
+
+    This is an example of double pendulum where we will do all kinematics
```

Comment:
```suggestion
    This is an example of simple double pendulum where we will do all kinematics
```

Reply:
The example that follows this does not seem to be a simple double pendulum. If you write "double pendulum" everyone is going to think of this: https://en.wikipedia.org/wiki/Double_pendulum and if you write "simple double pendulum" that means it only deals with point masses in the double pendulum.

I think showing a simple double pendulum is the best first example to show here.

Reply:
You can have this rod + plate, 90 degree axis pendulum too, but it needs a different name. It may also best belong in joint_example.rst page you plan to make.

Reply:
> You can have this rod + plate, 90 degree axis pendulum too, but it needs a different name. It may also best belong in joint_example.rst page you plan to make.

Okay, I'll change that.